src/graphics/device.js

```diff
@@ -2463,15 +2474,19 @@ pc.extend(pc, function () {
             if (shader !== this.shader) {
                 this.shader = shader;
 
-                if (! shader.ready)
-                    shader.link();
+                if (! shader.ready) {
```

Comment:
Would complicate observable logic to me. Link must never ever be called, if the shader is ready, and it makes it more clear, than a single line.